IMPORTING THE DATASET

MUYOBO ARNOLD M24B13/024
SHANTAL NABULO M24B13/008
OOKO BRENDAH M24B13/

In [1]:
import pandas as pd
loans = pd.read_csv("Loan_Approval_Data (1).csv")
loans

Loan_ID  Gender Married Dependents     Education Self_Employed  \
0    LP001002    Male      No          0      Graduate            No   
1    LP001003    Male     Yes          1      Graduate            No   
2    LP001005    Male     Yes          0      Graduate           Yes   
3    LP001006    Male     Yes          0  Not Graduate            No   
4    LP001008    Male      No          0      Graduate            No   
..        ...     ...     ...        ...           ...           ...   
609  LP002978  Female      No          0      Graduate            No   
610  LP002979    Male     Yes         3+      Graduate            No   
611  LP002983    Male     Yes          1      Graduate            No   
612  LP002984    Male     Yes          2      Graduate            No   
613  LP002990  Female      No          0      Graduate           Yes   

     ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
0               5849                0.0         NaN             360.0   
1               4583             1508.0       128.0             360.0   
2               3000                0.0        66.0             360.0   
3               2583             2358.0       120.0             360.0   
4               6000                0.0       141.0             360.0   
..               ...                ...         ...               ...   
609             2900                0.0        71.0             360.0   
610             4106                0.0        40.0             180.0   
611             8072              240.0       253.0             360.0   
612             7583                0.0       187.0             360.0   
613             4583                0.0       133.0             360.0   

     Credit_History Property_Area Loan_Status  
0               1.0         Urban           Y  
1               1.0         Rural           N  
2               1.0         Urban           Y  
3               1.0         Urban           Y  
4               1.0         Urban           Y  
..              ...           ...         ...  
609             1.0         Rural           Y  
610             1.0         Rural           Y  
611             1.0         Urban           Y  
612             1.0         Urban           Y  
613             0.0     Semiurban           N  

[614 rows x 13 columns]

Finding the datatypes

In [2]:
loans.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 614 entries, 0 to 613
Data columns (total 13 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Loan_ID            614 non-null    object 
 1   Gender             601 non-null    object 
 2   Married            611 non-null    object 
 3   Dependents         599 non-null    object 
 4   Education          614 non-null    object 
 5   Self_Employed      582 non-null    object 
 6   ApplicantIncome    614 non-null    int64  
 7   CoapplicantIncome  614 non-null    float64
 8   LoanAmount         592 non-null    float64
 9   Loan_Amount_Term   600 non-null    float64
 10  Credit_History     564 non-null    float64
 11  Property_Area      614 non-null    object 
 12  Loan_Status        614 non-null    object 
dtypes: float64(4), int64(1), object(8)
memory usage: 62.5+ KB


Finding marginal probability of applicant income

In [13]:
## Discretize the applicantincome into bins
bins = [0,5000,10000,15000,20000,25000,30000,35000,40000,45000,50000,55000,60000,65000,70000,75000,80000,85000]
labels = ['0-5000', '5000-10000', '10000-15000', '15000-20000', '20000-25000', '25000-30000', '30000-35000', '35000-40000', '40000-45000', '45000-50000','50000-55000','55000-60000', '60000-65000','65000-70000', '70000-75000', '75000-80000', '80000-85000']

#Creating a new column
api = pd.cut(loans['ApplicantIncome'], bins=bins, labels=labels, right=False)

# Calculating the marginal  probabilities
api_prob = api.value_counts(normalize=True).sort_index
api_prob

<bound method Series.sort_index of ApplicantIncome
0-5000         0.680782
5000-10000     0.229642
10000-15000    0.047231
15000-20000    0.022801
20000-25000    0.008143
35000-40000    0.004886
30000-35000    0.001629
60000-65000    0.001629
50000-55000    0.001629
80000-85000    0.001629
25000-30000    0.000000
40000-45000    0.000000
45000-50000    0.000000
55000-60000    0.000000
65000-70000    0.000000
70000-75000    0.000000
75000-80000    0.000000
Name: proportion, dtype: float64>

Finding marginal probability of property area

In [14]:
property_prob = loans['Property_Area'].value_counts()/len(loans)
print(f"The marginal probabilities of property_area are: \n {property_prob}")

The marginal probabilities of property_area are: 
 Property_Area
Semiurban    0.379479
Urban        0.328990
Rural        0.291531
Name: count, dtype: float64


The probability that a female applicant has her loan status reflecting Y

In [51]:
#make the events in Python
event_a = loans[loans['Gender']== 'Female']
event_b = loans[loans['Loan_Status'] == 'Y']

joint_event = loans[(loans['Gender'] == 'Female') & (loans['Loan_Status'] == 'Y')]

#get the jjoint probability
joint_prob = len(joint_event) / len(loans)
joint_prob

#getting the probability of the event_b
prob_event_b = len(event_b) / len(loans)

#getting the conditional probability
conditional_prob = joint_prob / prob_event_b
conditional_prob


The gender sample space is 614.
The event of getting a female is 112
The probability of getting a female is 0.18241042345276873
The probability of having a Y loan status is 0.6872964169381107
The probability of a female who has Y loan status is {joint_prob}
The probability of that a female applicant has her loan_status reflecting Y is 0.18241042345276873


Explaining the central tendency of any two continuous variables

In [29]:
loans.describe()

ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
count       614.000000         614.000000  592.000000         600.00000   
mean       5403.459283        1621.245798  146.412162         342.00000   
std        6109.041673        2926.248369   85.587325          65.12041   
min         150.000000           0.000000    9.000000          12.00000   
25%        2877.500000           0.000000  100.000000         360.00000   
50%        3812.500000        1188.500000  128.000000         360.00000   
75%        5795.000000        2297.250000  168.000000         360.00000   
max       81000.000000       41667.000000  700.000000         480.00000   

       Credit_History  
count      564.000000  
mean         0.842199  
std          0.364878  
min          0.000000  
25%          1.000000  
50%          1.000000  
75%          1.000000  
max          1.000000

For LoanAmount; The average value of loanAmount is 146.412162, the middle number for LoanAmount is 128
For CoapplicantIncome; the average value is 1621.245798 , the middle number is 1188.5

Explaining the relationship between the variables Gender and Loan_Status

In [47]:
CrosstabResult=pd.crosstab(index=loans['Gender'],columns=loans['Loan_Status'])
print(CrosstabResult)

from scipy.stats import chi2_contingency
ChiSqResult = chi2_contingency(CrosstabResult)
print('The P-Value is:', ChiSqResult[1])

Loan_Status    N    Y
Gender               
Female        37   75
Male         150  339
The P-Value is: 0.7086529816451106


The p_value, 0.7086529816451106 is greater than 0.05, therefore loan status is not correlated with gender
37 females and 150 males had their loan applications denied
75 females and 339 males had their loan applications approved

In [58]:
#make the events in Python
event_a = loans[loans['Gender']== 'Female']
event_b = loans[loans['Loan_Status'] == 'Y']

joint_event = loans[(loans['Gender'] == 'Female') & (loans['Loan_Status'] == 'Y')]

#get the jjoint probability
joint_prob = len(joint_event) / len(loans)
joint_prob

#getting the probability of the event_b
prob_event_b = len(event_b) / len(loans)

#getting the conditional probability
conditional_prob = joint_prob / prob_event_b
conditional_prob

0.17772511848341233